src/category_theory/limits/mono_coprod.lean

```diff
@@ -0,0 +1,114 @@
+/-
+Copyright (c) 2022 Joël Riou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Joël Riou
+-/
+
+import category_theory.limits.types
+import category_theory.morphism_property
+
+/-!
+
+# Categories where inclusions into coproducts are monomorphisms
+
+If `C` is a category that has finite coproducts, the class `mono_coprod C`
+expresses that left inclusions `A ⟶ A ⨿ B` are monomorphisms. If it is so,
+it is shown that right inclusions are also monomorphisms.
+
+TODO @joelriou: show that if `X : I → C` and `ι : J → I` is an injective map,
+then the canonical morphism `∐ (X ∘ ι) ⟶ ∐ X` is a monomorphism.
```

Comment:
Can we also have the todo that in a cartesian closed category with coproducts, `mono_coprod` holds?

Reply:
I could add it, but I am not sure why this is true?

Reply:
a ccc is distributive (which we also don't have yet unfortunately), and distributivity is enough for mono_coprod to be true: https://ncatlab.org/toddtrimble/published/distributivity+implies+monicity+of+coproduct+inclusions

Reply:
maybe a better todo is to define distributive categories, and show they're all mono_coprod - up to you though!

Reply:
This may not be my top priority, but I have added this TODO to the file...